clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -918,6 +999,44 @@ void LoopInfoStack::push(BasicBlock *Header, clang::ASTContext &Ctx,
       setSYCLMaxConcurrencyEnable();
       setSYCLMaxConcurrencyNThreads(ArgVal.getSExtValue());
     }
+
+    if (IntelFPGALoopCoalesce) {
+      llvm::APSInt ArgVal(32);
+      if (auto *LCE = IntelFPGALoopCoalesce->getNExpr()) {
+        bool IsValid = LCE->isIntegerConstantExpr(ArgVal, Ctx);
+        assert(IsValid && "Not an integer constant expression");
+        (void)IsValid;
+        setSYCLLoopCoalesceNLevels(ArgVal.getSExtValue());
+      } else {
+        setSYCLLoopCoalesceEnable();
+      }
+    }
+
+    if (IntelFPGADisableLoopPipelining) {
+      setSYCLLoopPipeliningDisable();
+    }
+
+    if (IntelFPGAMaxInterleaving) {
+      llvm::APSInt ArgVal(32);
+      bool IsValid =
+          IntelFPGAMaxInterleaving->getNExpr()->isIntegerConstantExpr(ArgVal,
+                                                                      Ctx);
+      assert(IsValid && "Not an integer constant expression");
+      (void)IsValid;
```

Comment:
Us writing compilers first and then having to fool them... apparently, to continue writing compilers! :)

Reply:
Unfortunately, `__attribute__((unused))` isn't supported by MSVC and `[[maybe_unused]]` attribute is a part of C++17 standard which is too modern for us.